src/components/select/filterable-select/filterable-select.test.tsx

```diff
@@ -83,33 +220,1577 @@ test("combobox has correct accessible name when aria-labelledby prop is provided
+      </FilterableSelect>
+    );
+    await user.type(screen.getByRole("combobox"), "gre");
+
+    expect(await screen.findByRole("option")).toHaveAttribute(
+      "aria-selected",
+      "true"
+    );
+  });
+
+  it("should not trim any following whitespace that matches an option text", async () => {
+    const user = userEvent.setup();
+    render(
+      <FilterableSelect label="filterable-select" onChange={() => {}} value="">
+        <Option value="opt1" text="black and white" />
+      </FilterableSelect>
+    );
+    const input = screen.getByRole("combobox") as HTMLInputElement;
+    await user.type(screen.getByRole("combobox"), "black ");
+
+    expect(input).toHaveValue("black and white");
+  });
+
+  it("should trim any following whitespace and select the matching option", async () => {
+    const user = userEvent.setup();
+    render(
+      <FilterableSelect label="filterable-select" onChange={() => {}} value="">
+        <Option value="opt1" text="green" />
+      </FilterableSelect>
+    );
+    await user.type(screen.getByRole("combobox"), "gre   ");
+
+    expect(await screen.findByRole("option")).toHaveAttribute(
+      "aria-selected",
+      "true"
+    );
+  });
+
+  it("should not trim the if only whitespace is typed", async () => {
+    const user = userEvent.setup();
+    render(
+      <FilterableSelect label="filterable-select" onChange={() => {}} value="">
+        <Option value="opt1" text="black" />
+      </FilterableSelect>
+    );
+    const input = screen.getByRole("combobox") as HTMLInputElement;
+    await user.type(screen.getByRole("combobox"), "   ");
+
+    expect(input).toHaveValue("   ");
+  });
+
+  it("should call the `onFilterChange` callback if passed", async () => {
+    const onFilterChangeFn = jest.fn();
+    const user = userEvent.setup();
+    render(
+      <FilterableSelect
+        label="filterable-select"
+        onChange={() => {}}
+        value=""
+        onFilterChange={onFilterChangeFn}
+      >
+        <Option value="opt1" text="red" />
+      </FilterableSelect>
+    );
+    await user.type(screen.getByRole("combobox"), "foo");
+
+    expect(onFilterChangeFn).toHaveBeenCalledWith("foo");
+  });
+
+  it("should not be call the `onFilterChange` callback if passed when the component rerenders", () => {
+    const onFilterChangeFn = jest.fn();
+    const { rerender } = render(
+      <FilterableSelect
+        label="filterable-select"
+        onChange={() => {}}
+        value=""
+        onFilterChange={onFilterChangeFn}
+      >
+        <Option value="opt1" text="red" />
+      </FilterableSelect>
+    );
+
+    rerender(
+      <FilterableSelect
+        label="filterable-select"
+        onChange={() => {}}
+        value=""
+        onFilterChange={onFilterChangeFn}
+      >
+        <Option value="opt1" text="red" />
+      </FilterableSelect>
+    );
+
+    expect(onFilterChangeFn).not.toHaveBeenCalled();
+  });
+
+  it("foo", async () => {
```

Comment:
**question:** was this left by accident?

Reply:
yep 🤦